release.csx

```diff
@@ -0,0 +1,84 @@
+#r ".\packages\Octokit.0.26.0\lib\net45\Octokit.dll"
+#r ".\packages\system.runtime.interopservices.runtimeinformation\4.0.0\lib\net45\System.Runtime.InteropServices.RuntimeInformation.dll"
+
+using Octokit;
+
+const string repoOwner = "FakeItEasy";
+const string repoName = "FakeItEasy";
+const string remote = "upstream";
+
+var version = Args.FirstOrDefault();
+if (version == null)
+{
+    throw new Exception("No version supplied");
+}
+
+var releaseBranchName = $"release/{version}";
+
+var token = ReadTokenFromNetrc();
+
+var gitHubClient = new GitHubClient(new ProductHeaderValue("FakeItEasy-build-scripts"));
+gitHubClient.Credentials = new Credentials(token);
+Cmd("git", $"checkout -b {releaseBranchName}");
+Cmd("git", $"push {remote} {releaseBranchName}");

```

Comment:
The output from these commands is dumped to the console, so it's not super pretty. Then again, convenient if something goes wrong.

Note we're assuming that the caller is already using the proper branch.

Reply:
> Note we're assuming that the caller is already using the proper branch.

It's a pretty big assumption... I think we should checkout `develop` first (or use `checkout -b {releaseBranchName} develop`).

Reply:
Can do. Assumes user's local `develop` is up to date.
If that's a concern, we could `checkout -b {releaseBranchName} {remote}/develop`. Although that seems to set tracking to the remote branch. I'm surprised.

Reply:
> Can do. Assumes user's local develop is up to date.
> If that's a concern, we could checkout -b {releaseBranchName} {remote}/develop.

But `{remote}/develop` could also be out of date... to be sure, you'd need to `git fetch` first.

> Although that seems to set tracking to the remote branch. I'm surprised.

Yes, it does. `--no-track` should prevent this.